plugins/inputs/cisco_telemetry_mdt/cisco_telemetry_mdt.go

```diff
@@ -430,11 +1500,26 @@ func (c *CiscoTelemetryMDT) parseContentField(grouper *metric.SeriesGrouper, fie
 
 	var nxAttributes, nxChildren, nxRows *telemetry.TelemetryField
 	isNXOS := !strings.ContainsRune(path, ':') // IOS-XR and IOS-XE have a colon in their encoding path, NX-OS does not
+	isEVENT := isNXOS && strings.Contains(path, "EVENT-LIST")
 	for _, subfield := range field.Fields {
 		if isNXOS && subfield.Name == "attributes" && len(subfield.Fields) > 0 {
 			nxAttributes = subfield.Fields[0]
 		} else if isNXOS && subfield.Name == "children" && len(subfield.Fields) > 0 {
-			nxChildren = subfield
+			if !isEVENT {
+				nxChildren = subfield
+			} else {
+				sub := subfield.Fields
+				if sub[0] != nil && sub[0].Fields[0].Name == "subscriptionId" && len(sub[0].Fields) >= 2 {
+					nxAttributes = sub[0].Fields[1].Fields[0].Fields[0].Fields[0].Fields[0].Fields[0]
+				}
+			}
+			//if nxAttributes == NULL then class based query.
+			if nxAttributes == nil {
```

Comment:
Please initialize `nxChildren` as `nil` as in case `isEVENT == true` and `if sub[0] != nil && sub[0].Fields[0].Name == "subscriptionId" && len(sub[0].Fields) >= 2` being `false`  this variable is not set and might contain values from the previous loop-cycle.

Reply:
this is still relevant